python/paddle/fluid/contrib/slim/quantization/qat2_int8_mkldnn_pass.py

```diff
@@ -193,6 +195,24 @@ def _set_dummy_fc_out_scales(self, graph):
                     # use input scale as a "dummy" scale
                     self._var_quant_scales[
                         output_name] = self._var_quant_scales[input_name]
+            elif op.name() in self._conv_ops:
+                input_name = op.input("Input")[0]
+                output_name = op.output("Output")[0]
+                if input_name in self._var_quant_scales and \
+                    output_name not in self._var_quant_scales:
+                    # use input scale as a "dummy" scale
+                    self._var_quant_scales[
+                        output_name] = self._var_quant_scales[input_name]
+            elif op.name() in self._matmul_ops:
+                input_x_name = op.input("X")[0]
+                input_y_name = op.input("Y")[0]
+                output_name = op.output("Out")[0]
+                if input_x_name in self._var_quant_scales and \
+                    input_y_name in self._var_quant_scales and \
+                    output_name not in self._var_quant_scales:
+                    # use input scale as a "dummy" scale
+                    self._var_quant_scales[
+                        output_name] = self._var_quant_scales[input_x_name]
```

Comment:
Pretty confident this could be simplifed to something like:
```python
def _set_scale(self, op_type, accepted_types, input_names, output_name):
    if op_type in accepted_types:
        should_set = True
        for input in input_names:
            has_in_scale = op.input(input)[0] in self._var_quant_scales
            should_set = should_set and has_in_scale
        has_out_scale =  op.output(output_name)[0] in self._var_quant_scales
        should_set = should_set and not has_out_scale
        if should_set:
            out_var_name = op.output(output_name)[0]
            in_var_name = op.input(input_names[0])[0]
            self._var_quant_scales[
                        out_var_name] = self._var_quant_scales[in_var_name]

for op in graph.all_op_nodes():                        
    _set_scale(op.name(), self._fc_ops, ["Input"], "Out")
    _set_scale(op.name(), self._conv_ops, ["Input"], "Output")
    _set_scale(op.name(), self._matmul_ops, ["X", "Y"], "Out")
```
(untested)

Reply:
Good point. Done.